src/styles/shellbar.scss

```diff
@@ -105,61 +105,61 @@ $block: #{$fd-namespace}-shellbar;
     }
   }
 
-  &__action {
+  .#{$block}__group .#{$block}__action {
     @include fd-reset();
     @include fd-set-margin-left($fd-shellbar-item-spacing);
 
     > * {
       vertical-align: middle;
     }
+  }
 
-    .#{$block}__input-group {
-      background: transparent;
-      background-color: var(--fdShellbar_Input_Background);
-      border: var(--fdShellbar_Input_Border);
-      border-radius: var(--fdShellbar_Input_Border_Radius);
-      cursor: text;
+  .#{$block}__group .#{$block}__input-group {
+    background: transparent;
+    background-color: var(--fdShellbar_Input_Background);
+    border: var(--fdShellbar_Input_Border);
+    border-radius: var(--fdShellbar_Input_Border_Radius);
+    cursor: text;
 
-      @include fd-hover() {
-        border-color: var(--fdShellbar_Input_Hover_Border_Color);
-        box-shadow: var(--sapContent_Interaction_Shadow);
-      }
+    @include fd-hover() {
+      border-color: var(--fdShellbar_Input_Hover_Border_Color);
+      box-shadow: var(--sapContent_Interaction_Shadow);
+    }
 
-      &-input {
-        border: none;
-        color: var(--fdShellbar_Input_Color);
-        position: relative;
-
-        @include fd-hover() {
-          &,
-          & + .#{$block}__input-group-addon {
-            background: transparent;
-            background-color: var(--fdShellbar_Input_Hover_Background) !important;
-          }
-        }
+    &-input {
+      border: none;
+      color: var(--fdShellbar_Input_Color);
+      position: relative;
 
-        &::placeholder {
-          color: var(--fdShellbar_Input_Placeholder_Color);
-          font-style: italic;
+      @include fd-hover() {
+        &,
+        & + .#{$block}__input-group-addon {
+          background: transparent;
+          background-color: var(--fdShellbar_Input_Hover_Background) !important;
         }
+      }
 
-        &--select {
-          font-style: normal;
-        }
+      &::placeholder {
+        color: var(--fdShellbar_Input_Placeholder_Color);
+        font-style: italic;
       }
 
-      &-addon .#{$block}__button {
-        height: var(--fdShellbar_Input_Addon_Dimension);
-        min-width: var(--fdShellbar_Input_Addon_Dimension);
-        border-radius: var(--fdShellbar_Input_Border_Radius);
-        border-width: var(--fdShellbar_Input_Addon_Border_Width);
+      &--select {
+        font-style: normal;
       }
     }
+
+    &-addon .#{$block}__button {
+      height: var(--fdShellbar_Input_Addon_Dimension);
+      min-width: var(--fdShellbar_Input_Addon_Dimension);
+      border-radius: var(--fdShellbar_Input_Border_Radius);
+      border-width: var(--fdShellbar_Input_Addon_Border_Width);
+    }
   }
 
-  .#{$block}__button {
+  .#{$block}__group .#{$block}__button {
     outline-color: var(--fdShellbar_Button_Outline_Color);
-    color: var(--sapShell_InteractiveTextColor);
+    color: var(--sapShell_InteractiveTextColor) !important;
```

Comment:
That's not ok. You have 3 levels of nesting and still needing `!important`?

Reply:
That basically make this element impossible to customize by the client.

Reply:
I'm fully agree with your arguments. However, to remove importants from this file we should remove them from some other files too. Yea, a total mess. 

Unfortunately, if we remove them somewhere it may lead to unexpected visual changes. And now we cannot catch them as we're out of limit in Chromatic.

I've opened a ticket https://github.com/SAP/fundamental-styles/issues/3420 so let's process this one once we have Chromatic to ensure we won't break anything.